src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -568,7 +617,9 @@ public long importKey(long hSession, CK_ATTRIBUTE[] attributes) throws PKCS11Exc
                 byte[] wrappedBytes = wrapCipher.doFinal(keyBytes);
 
                 // Unwrap the secret key.
-                CK_ATTRIBUTE[] unwrapAttributes = token.getAttributes(TemplateManager.O_IMPORT, keyClass, keyType, attributes);
+                // Need additional attributes for EC private key.
+                attributes = new CK_ATTRIBUTE[ckAttrsMap.size()];
+                CK_ATTRIBUTE[] unwrapAttributes = token.getAttributes(TemplateManager.O_IMPORT, keyClass, keyType, ckAttrsMap.values().toArray(attributes));
```

Comment:
Please don't modify parameters; and it's not necessary here:
```java
                CK_ATTRIBUTE[] unwrapAttributes = token.getAttributes(
                        TemplateManager.O_IMPORT,
                        keyClass,
                        keyType,
                        ckAttrsMap.values().toArray(new CK_ATTRIBUTE[ckAttrsMap.size()]));
```

Reply:
edited